src/kernel/bitcoin-chainstate.cpp

```diff
@@ -0,0 +1,200 @@
+#include <kernel/bitcoinkernel_wrapper.h>
+
+#include <cassert>
+#include <filesystem>
+#include <iostream>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <sstream>
+
+std::vector<unsigned char> hex_string_to_char_vec(const std::string& hex)
+{
+    std::vector<unsigned char> bytes;
+
+    for (size_t i{0}; i < hex.length(); i += 2) {
+        std::string byteString{hex.substr(i, 2)};
+        unsigned char byte = (char)std::strtol(byteString.c_str(), nullptr, 16);
```

Comment:
i would really prefer not to bring back use of `strtol` in C++ code; it has some known issues with locale-dependence (especially on Linux). what about:

```c++
#include <charconv>
...
std::vector<unsigned char> hex_string_to_char_vec(const std::string& hex)
{
    std::vector<unsigned char> bytes;

    for (size_t i{0}; i < hex.length(); i += 2) {
        unsigned int val{0};
        auto [p, ec] = std::from_chars(hex.data() + i, hex.data() + i + 2, val, 16);
        if (ec == std::errc{} && p == hex.data() + i + 2) {
            bytes.push_back(val);
        }
    }

    return bytes;
}
```